Quantum-Control-Applications/Superconducting/Single Flux Tunable Transmon/SSB mixer for readout/configuration.py

```diff
@@ -0,0 +1,323 @@
+import numpy as np
+from scipy.signal.windows import gaussian
+from qualang_tools.config.waveform_tools import drag_gaussian_pulse_waveforms
+
+
+#######################
+# AUXILIARY FUNCTIONS #
+#######################
+
+# IQ imbalance matrix
+def IQ_imbalance(g, phi):
+    """
+    Creates the correction matrix for the mixer imbalance caused by the gain and phase imbalances, more information can
+    be seen here:
+    https://docs.qualang.io/libs/examples/mixer-calibration/#non-ideal-mixer
+    :param g: relative gain imbalance between the I & Q ports. (unit-less), set to 0 for no gain imbalance.
+    :param phi: relative phase imbalance between the I & Q ports (radians), set to 0 for no phase imbalance.
+    """
+    c = np.cos(phi)
+    s = np.sin(phi)
+    N = 1 / ((1 - g**2) * (2 * c**2 - 1))
+    return [float(N * x) for x in [(1 - g) * c, (1 + g) * s, (1 - g) * s, (1 + g) * c]]
+
+
+# Live plotting
+def interrupt_on_close(figure, current_job):
+    def on_close(event):
+        print("Execution stopped by user!")
+        current_job.halt()
+        event.canvas.stop_event_loop()
+
+    figure.canvas.mpl_connect("close_event", on_close)
+
+
+#############
+# VARIABLES #
+#############
+
+qop_ip = "10.0.2.1"
+qop_port = 80
+
+# Qubits
+qubit_LO = 7.4e9  # Used only for mixer correction and frequency rescaling for plots or computation
+qubit_IF = 110e6
+mixer_qubit_g = 0.0
+mixer_qubit_phi = 0.0
+
+qubit_T1 = int(10e3)
+
+const_len = 100
+const_amp = 50e-3
+
+pi_len = 100
+pi_amp = 0.05
+
+gauss_len = 200
+gauss_sigma = gauss_len / 5
+gauss_amp = 0.25
+gauss_wf = gauss_amp * gaussian(gauss_len, gauss_sigma)
+
+x180_len = 100
+x180_sigma = x180_len / 5
+x180_amp = 0.15
+x180_drag_wf, x180_drag_der_wf = np.array(
+    drag_gaussian_pulse_waveforms(x180_amp, x180_len, x180_sigma, alpha=0, delta=1)
+)
+# No DRAG when alpha=0, it's just a gaussian.
+
+x90_len = x180_len
+x90_sigma = x90_len / 5
+x90_amp = x180_amp / 2
+x90_drag_wf, x90_drag_der_wf = np.array(drag_gaussian_pulse_waveforms(x90_amp, x90_len, x90_sigma, alpha=0, delta=1))
+# No DRAG when alpha=0, it's just a gaussian.
+
+# Resonator
+resonator_LO = 4.8e9  # Used only for mixer correction and frequency rescaling for plots or computation
+resonator_IF = 60e6
+mixer_resonator_g = 0.0
+mixer_resonator_phi = 0.0
+
+readout_len = 2000
+readout_amp = 0.25
+
+time_of_flight = 300
+
+# Flux line
+flux_line_IF = 0e6
```

Comment:
Just for generality sake, since I don't know much about these experiments I tried to keep the config as general as possible. But if you tell me that people will never send any IF to the flux line, then I can remove it...

Reply:
Yes, there's never an IF in the flux line.
Moreover, this means we can remove the "intermediate frequency" line from the element (it means that there won't be an oscillator associated with it)

Reply:
Alright, I'll remove it!